waku/api.go

```diff
@@ -216,6 +218,7 @@ type NewMessage struct {
 	TargetPeer string           `json:"targetPeer"`
 }
 
+// TODO if this is dead code why not remove it? I've checked and there are no usages of this
 type newMessageOverride struct { // nolint: deadcode,unused
 	PublicKey hexutil.Bytes
 	Payload   hexutil.Bytes
```

Comment:
I feel this code just needs to be removed. I've checked it and there are no usages in the code base.

Reply:
if not used I guess it's ok to remove, it's probably some legacy bit

Reply:
Kill it.